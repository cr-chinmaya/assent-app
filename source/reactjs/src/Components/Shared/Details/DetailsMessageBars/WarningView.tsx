import * as React from 'react';
import * as Styled from './DetailsMessageBarsStyling';
import { Stack } from '@fluentui/react/lib/Stack';
import { MessageBar, MessageBarType } from '@fluentui/react';
import { CollapsibleSection } from '../../Components/CollapsibleSection';

interface WarningViewProps {
    warningTitle?: string;
    warningMessages: string[] | null;
    onDismiss?: VoidFunction | null;
    isCollapsible?: boolean;
    messageBarStyles?: any;
}

const WarningView = ({
    warningTitle = '',
    warningMessages = null,
    onDismiss = null,
    isCollapsible = false,
    messageBarStyles = null
}: WarningViewProps): JSX.Element => {
    const messageElements = warningMessages?.map((item: string, index: number) => (
        <Stack.Item styles={warningMessages.length > 1 && Styled.WarningViewStackStylesBottomBorder} key={index}>
            <p>{item}</p>
        </Stack.Item>
    ));

    return isCollapsible ? (
        <CollapsibleSection
            defaultIsExpanded={false}
            titleText={warningTitle ? warningTitle : ''}
            renderHeaderAs="div"
            style={Styled.CollapsibleStyle}
        >
            <Stack tokens={Styled.WarningViewStackTokensLargeGap} style={Styled.CollapsibleStyle}>
                {messageElements}
            </Stack>
        </CollapsibleSection>
    ) : (
        <MessageBar
            messageBarType={MessageBarType.warning}
            isMultiline={true}
            aria-label={'Warning message'}
            onDismiss={onDismiss}
            styles={messageBarStyles}
        >
            <Stack tokens={Styled.WarningViewStackTokensLargeGap}>
                <>
                    {warningTitle && (
                        <Stack.Item>
                            <Styled.DetailsMessageBarTitle>{warningTitle}</Styled.DetailsMessageBarTitle>
                        </Stack.Item>
                    )}
                    {messageElements}
                </>
            </Stack>
        </MessageBar>
    );
};

export default WarningView;
