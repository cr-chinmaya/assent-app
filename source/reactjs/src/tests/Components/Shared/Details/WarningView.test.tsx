import * as React from 'react';
import { render} from '@testing-library/react';
import WarningView from '../../../../Components/Shared/Details/DetailsMessageBars/WarningView'

describe('WarningView', () => {
    it('should render Warning view correctly with warning title', () => {
        const wrapper = render(
            <WarningView
            warningTitle={''}
            warningMessages= {null}
            onDismiss={null}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Warning view correctly with undefined props', () => {
        const wrapper = render(
            <WarningView
            warningTitle={undefined}
            warningMessages= {undefined}
            onDismiss={undefined}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Warning view correctly when onDismiss and warningMessages is undefined', () => {
        const wrapper = render(
            <WarningView
            warningTitle={"warningTitle"}
            warningMessages= {undefined}
            onDismiss={undefined}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Warning view correctly when warningMessage is defined' , () => {
        const wrapper = render(
            <WarningView
            warningTitle={"warningTitle"}
            warningMessages= {['A','B']}
            onDismiss={undefined}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});