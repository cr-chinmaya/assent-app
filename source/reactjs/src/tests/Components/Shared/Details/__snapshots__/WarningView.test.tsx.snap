// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningView should render Warning view correctly when onDismiss and warningMessages is undefined 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ms-MessageBar ms-MessageBar--warning ms-MessageBar-multiline root-84"
      >
        <div
          class="ms-MessageBar-content content-85"
        >
          <div
            aria-hidden="true"
            class="ms-MessageBar-icon iconContainer-86"
          >
            <i
              aria-hidden="true"
              class="icon-93"
              data-icon-name="Info"
            />
          </div>
          <div
            aria-live="polite"
            class="ms-MessageBar-text text-88"
            id="MessageBar2"
            role="status"
          >
            <span
              aria-label="Warning message"
              class="ms-MessageBar-innerText innerText-89"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ms-MessageBar ms-MessageBar--warning ms-MessageBar-multiline root-84"
    >
      <div
        class="ms-MessageBar-content content-85"
      >
        <div
          aria-hidden="true"
          class="ms-MessageBar-icon iconContainer-86"
        >
          <i
            aria-hidden="true"
            class="icon-93"
            data-icon-name="Info"
          />
        </div>
        <div
          aria-live="polite"
          class="ms-MessageBar-text text-88"
          id="MessageBar2"
          role="status"
        >
          <span
            aria-label="Warning message"
            class="ms-MessageBar-innerText innerText-89"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WarningView should render Warning view correctly when warningMessage is defined 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ms-MessageBar ms-MessageBar--warning ms-MessageBar-multiline root-84"
      >
        <div
          class="ms-MessageBar-content content-85"
        >
          <div
            aria-hidden="true"
            class="ms-MessageBar-icon iconContainer-86"
          >
            <i
              aria-hidden="true"
              class="icon-93"
              data-icon-name="Info"
            />
          </div>
          <div
            aria-live="polite"
            class="ms-MessageBar-text text-88"
            id="MessageBar3"
            role="status"
          >
            <span
              aria-label="Warning message"
              class="ms-MessageBar-innerText innerText-89"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ms-MessageBar ms-MessageBar--warning ms-MessageBar-multiline root-84"
    >
      <div
        class="ms-MessageBar-content content-85"
      >
        <div
          aria-hidden="true"
          class="ms-MessageBar-icon iconContainer-86"
        >
          <i
            aria-hidden="true"
            class="icon-93"
            data-icon-name="Info"
          />
        </div>
        <div
          aria-live="polite"
          class="ms-MessageBar-text text-88"
          id="MessageBar3"
          role="status"
        >
          <span
            aria-label="Warning message"
            class="ms-MessageBar-innerText innerText-89"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WarningView should render Warning view correctly with undefined props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ms-MessageBar ms-MessageBar--warning ms-MessageBar-multiline root-84"
      >
        <div
          class="ms-MessageBar-content content-85"
        >
          <div
            aria-hidden="true"
            class="ms-MessageBar-icon iconContainer-86"
          >
            <i
              aria-hidden="true"
              class="icon-93"
              data-icon-name="Info"
            />
          </div>
          <div
            aria-live="polite"
            class="ms-MessageBar-text text-88"
            id="MessageBar1"
            role="status"
          >
            <span
              aria-label="Warning message"
              class="ms-MessageBar-innerText innerText-89"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ms-MessageBar ms-MessageBar--warning ms-MessageBar-multiline root-84"
    >
      <div
        class="ms-MessageBar-content content-85"
      >
        <div
          aria-hidden="true"
          class="ms-MessageBar-icon iconContainer-86"
        >
          <i
            aria-hidden="true"
            class="icon-93"
            data-icon-name="Info"
          />
        </div>
        <div
          aria-live="polite"
          class="ms-MessageBar-text text-88"
          id="MessageBar1"
          role="status"
        >
          <span
            aria-label="Warning message"
            class="ms-MessageBar-innerText innerText-89"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`WarningView should render Warning view correctly with warning title 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ms-MessageBar ms-MessageBar--warning ms-MessageBar-multiline root-84"
      >
        <div
          class="ms-MessageBar-content content-85"
        >
          <div
            aria-hidden="true"
            class="ms-MessageBar-icon iconContainer-86"
          >
            <i
              aria-hidden="true"
              class="icon-93"
              data-icon-name="Info"
            />
          </div>
          <div
            aria-live="polite"
            class="ms-MessageBar-text text-88"
            id="MessageBar0"
            role="status"
          >
            <span
              aria-label="Warning message"
              class="ms-MessageBar-innerText innerText-89"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ms-MessageBar ms-MessageBar--warning ms-MessageBar-multiline root-84"
    >
      <div
        class="ms-MessageBar-content content-85"
      >
        <div
          aria-hidden="true"
          class="ms-MessageBar-icon iconContainer-86"
        >
          <i
            aria-hidden="true"
            class="icon-93"
            data-icon-name="Info"
          />
        </div>
        <div
          aria-live="polite"
          class="ms-MessageBar-text text-88"
          id="MessageBar0"
          role="status"
        >
          <span
            aria-label="Warning message"
            class="ms-MessageBar-innerText innerText-89"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
